### PR TITLE
40K: Seeker of Slaanesh and support

### PR DESCRIPTION
**Pull request by Northmoc**
I think this has Trove working again...

Seeker isn't working yet, though... ideas?

### CODE REVIEW

File: forge-game/src/main/java/forge/game/combat/GlobalAttackRestrictions.java [hunk 1]
```diff
@@ -19,9 +20,11 @@
 public class GlobalAttackRestrictions {
 
     private final int max;
+    private final int min;
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/combat/GlobalAttackRestrictions.java:23]: This looks wrong, it's a requirement but you're trying to model it as a restriction

_Trove of Temptation_ does that better, but we need to research if it's a single requirement on a player or on every creature

it matters when comparing which attack is legal

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/combat/GlobalAttackRestrictions.java:23]: Ah, I see the distinction.

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/combat/GlobalAttackRestrictions.java:23]: It definitely reads like it's a requirement on the player... there is a note that says the Trove also is:
image

File: forge-game/src/main/java/forge/game/card/Card.java [hunk 2]
```diff
@@ -6350,6 +6350,7 @@ public void onEndOfCombat(final Player active) {
             chosenModesYourCombat.clear();
             chosenModesYourCombatStatic.clear();
             updateAbilityTextForView();
+            active.setMustAttack(false);
```
Comment (Northmoc) [inline forge-game/src/main/java/forge/game/card/Card.java:6353]: is there a better place to do this?

File: forge-game/src/main/java/forge/game/combat/AttackConstraints.java [hunk 3]
```diff
@@ -405,7 +412,11 @@ public final int countViolations(final Map<Card, GameEntity> attackers) {
         }
 
         int violations = 0;
+
         for (final Card possibleAttacker : possibleAttackers) {
+            if (possibleAttacker.getController().mustAttack() && attackers.size() < 1) {
+                violations++;
```
Comment (Northmoc) [inline forge-game/src/main/java/forge/game/combat/AttackConstraints.java:418]: the AI doesn't seem to care if it has a violation or two – I think I am misunderstanding what this code does

I also tried `return -1;` here which puts Forge into a forever loop of `AI Attack declaration invalid, defaulting to: {}` when the AI doesn't want to attack

**Pull request closed without merging**